Tests/Extensions/XCTestExtensions.swift

```diff
@@ -11,28 +11,29 @@ import XCTest
 extension XCTestCase {
     
     /// Checks for the callback to be the expected value within the given timeout.
+    /// Awaiting result using a Task.sleep.
+    /// - Note: See `waitForConditionUsingMainActor` to validate the condition on the main actor.
     ///
     /// - Parameters:
     ///   - condition: The condition to check for.
     ///   - timeout: The timeout in which the callback should return true.
     ///   - description: A string to display in the test log for this expectation, to help diagnose failures.
-    func waitFor(_ condition: @autoclosure () -> Bool, timeout: TimeInterval, description: String) {
-        let expectation = self.expectation(description: description)
-        
-        let end = Date().addingTimeInterval(timeout)
-        
-        while !condition() && 0 < end.timeIntervalSinceNow {
-            if RunLoop.current.run(mode: RunLoop.Mode.default, before: Date(timeIntervalSinceNow: 0.002)) {
-                Thread.sleep(forTimeInterval: 0.002)
-            }
-        }
-        
-        if !condition() {
-            XCTFail("Timed out waiting for condition to be true")
-        } else {
-            expectation.fulfill()
-        }
+    func waitForCondition(
```

Comment:
This concurrency supporting variant is needed to ensure expectations fulfill correctly while running async tasks.